confgenerator/otel/modular.go

```diff
@@ -143,13 +147,13 @@ func (c ModularConfig) Generate() (string, error) {
 		"service":    service,
 	}
 
-	resourceDetectionProcessors := map[ResourceDetectionMode]Component{
-		Override: GCPResourceDetector(true),
-		Upsert:   GCPResourceDetector(false),
-	}
-	resourceDetectionProcessorNames := map[ResourceDetectionMode]string{
-		Override: resourceDetectionProcessors[Override].name("_global_0"),
-		Upsert:   resourceDetectionProcessors[Upsert].name("_global_1"),
+	if r, ok := resource.(resourcedetector.BMSResource); ok {
+		processors["resource/bms"] = AddResourceAttributes(map[string]string{
```

Comment:
Do you have any suggestions for an easy way to detect when to not override resource attributes? Or, should I just create a static list of receivers for which we don't want to modify attributes?

Reply:
That part is already done in this code (the two cases of `Override` and `Upsert`) - just reuse the same structure.

Reply:
More specifically, move this up to line 157, and replace the `resourceDetectionProcessors` map with two new processors. Then you also don't need the added processor at line 198.

Also, this doesn't need to be BMS-specific. I think I would add a `OTelResourceAttributes() map[string]string` method to the `Resource` interface, which you can just call from here.

Reply:
I apologize for such simple questions, I'm still new to the opentelemetry world, but can you provide an example when we would not want to override existing resource attributes ? Is this primarily needed for cases where libraries are sending telemetry data to a collector running on a different host ?

I'm unsure how to reuse the same structure because, unlike the resource detection processor, the resource processor doesn't support the "override" parameter.

Reply:
Please ignore the last part of my previous comment. I think I got it now. I didn't see your previous comment when I posted mine.

Reply:
Done. PTAL ?